### PR TITLE
Added a new kick system based on buttons, removed the old one

### PR DESCRIPTION
**Pull request by AmyrAhmady**

### CODE REVIEW

File: lobby.go [hunk 1]
```diff
@@ -207,6 +207,50 @@ func wsListen(lobby *Lobby, player *Player, socket *websocket.Conn) {
 						lobby.Drawer.WriteAsJSON(JSEvent{Type: "your-turn"})
 					}
 				}
+			} else if received.Type == "on-kick-button" {
```
Comment (Bios-Marcel) [inline lobby.go]: Button is an implementation detail, just call it `kick-vote`

File: lobby.go [hunk 2]
```diff
@@ -207,6 +207,50 @@ func wsListen(lobby *Lobby, player *Player, socket *websocket.Conn) {
...
+				toKickID, isString := (received.Data).(string)
+				toKick := -1
+				if !isString {
+					fmt.Println("Invalid data")
+					continue
+				}
+
+				for index, otherPlayer := range lobby.Players {
+					if otherPlayer.ID == toKickID {
+						toKick = index
```
Comment (Bios-Marcel) [inline lobby.go]: add a break condition here for better performance. There is no need to keep iterating if we have already found the player.

File: lobby.go [hunk 3]
```diff
@@ -207,6 +207,50 @@ func wsListen(lobby *Lobby, player *Player, socket *websocket.Conn) {
...
+				if toKick != -1 {
+					if player.votedForKick[toKickID] {
+						continue
+					}
+
+					player.votedForKick[toKickID] = true
+					playerToKick := lobby.Players[toKick]
+					playerToKick.voteKickCount++
+
+					if playerToKick.voteKickCount > 2 {
```
Comment (Bios-Marcel) [inline lobby.go]: We might want to make this somewhat dependent on the amount of players. Like what if you have 24 people and two of them could kick everyone?

File: lobby.go [hunk 4]
```diff
@@ -207,6 +207,50 @@ func wsListen(lobby *Lobby, player *Player, socket *websocket.Conn) {
...
+						if lobby.Drawer == playerToKick {
+							endRound(lobby)
+						}
+						isPlayerToKickTheLobbyOwner := (lobby.Owner == playerToKick)
+						player.votedForKick[toKickID] = false
```
Comment (Bios-Marcel) [inline lobby.go]: You only reset this for the last player that voted, the others will still have the trash in their `voteForKick` map.

Comment (Bios-Marcel) [reply lobby.go]: You could also remove the entry instead of setting it to false: `remove(player.votedForKick, toKickID)`

File: lobby.go [hunk 5]
```diff
@@ -317,35 +359,6 @@ func commandStart(caller *Player, lobby *Lobby) {
 	}
 }
 
-func commandKick(caller *Player, lobby *Lobby) {
```
Comment (Bios-Marcel) [inline lobby.go]: Keep this method and put the code from above in it.

File: lobby.html [hunk 6]
```diff
@@ -311,6 +311,10 @@
             wordDialog.style.visibility = "hidden";
         }
 
+        function onClickKickButton(playerId) {
+            socket.send(JSON.stringify({ Type: "on-kick-button", Data: playerId }));
```
Comment (Bios-Marcel) [inline lobby.html]: rename the event here as well

File: lobby_logic.go [hunk 7]
```diff
@@ -25,14 +25,18 @@ type Player struct {
 	// own socket.
 	socketMutex *sync.Mutex
 
+	// ID uniquely identified the Lobby.
```
Comment (Bios-Marcel) [inline lobby_logic.go]: This is the player, not the lobby, this should be `ID uniquely identified the Player`

File: resources/lobby_players.css [hunk 8]
```diff
@@ -29,6 +29,18 @@
     font-size: 1.5rem;
 }
 
+.kick-button {
+    border: 0;
+    padding: 2px;
+    background-color: rgb(218, 218, 218)t;
```
Comment (Bios-Marcel) [inline resources/lobby_players.css]: bruh, there is a `t` in the end

**Comment by AmyrAhmady** [comment]
a screenshot of how it looks like:
aaaaa

**Comment by Bios-Marcel** [comment]
Kick votes should be printed to the chat, however, the player should stay anonymous:

Example Chat output:

Marcel: Cat
Amir: Dog
**(1/4) players voted to kick Kevin**
Amir: egg

Where `1` is the amount of players that currently vote to kick kevin and `4` the amount of necessary players, not total players.

**Pull request merged**